tests/testing-framework/src/runtimes/in_process_spin.rs

```diff
@@ -52,9 +52,52 @@ impl InProcessSpin {
 }
 
 impl Runtime for InProcessSpin {
-    type Config = ();
-
     fn error(&mut self) -> anyhow::Result<()> {
         Ok(())
     }
 }
+
+impl InProcessSpin {
+    pub fn config(
+        services_config: ServicesConfig,
+        preboot: impl FnOnce(&mut TestEnvironment<InProcessSpin>) -> anyhow::Result<()> + 'static,
+    ) -> TestEnvironmentConfig<Self> {
+        TestEnvironmentConfig {
+            services_config,
+            create_runtime: Box::new(|env| {
+                preboot(env)?;
+                tokio::runtime::Runtime::new()
+                    .context("failed to start tokio runtime")?
+                    .block_on(async {
+                        use spin_trigger::{
```

Comment:
This is pretty deep; maybe un-nest this `async` block into an `async fn`? :shrug: